client/src/util/dataframe/dict-encoded.ts

```diff
@@ -0,0 +1,23 @@
+/*
+This module contains helper functions to map an array of categorical codes to values
+*/
+import { DictEncodedArray } from "../../common/types/arraytypes";
+import { DataframeValueArray } from "./types";
+
+export function mapCodesToValues(
+  column: DictEncodedArray,
+  __id: string
+): DataframeValueArray {
+  const result: DataframeValueArray = new Array(column.length);
+  for (const [i, _] of column.entries()) {
+    result[i] = column.vat(i);
+  }
+  return result;
+}
+
+export function hashMapCodesToValues(
```

Comment:
this function appears to be only used as a memoization key in dataframe.ts.  It can potentially create a huge string which seems inefficient.

Any reason you can't just stash a unique (small) ID inside the VAT array (like we already do for Dataframe objects), and then use that as the memoization key?

For example, each of the `DictEncoded*Array` classes could have a class variable (number) that is used to seed a globally unique ID which is set in the `codeMapping` object (inside the `setCodeMapping` method).